src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -183,7 +184,73 @@ class EventsRelation(config: RelationConfig)(@transient val sqlContext: SQLConte
     (postUpdate, postNewItems).parMapN((_, _) => ())
   }
 
-  override val fieldsWithPushdownFilter: Seq[String] = Seq("source", "type")
+  override val fieldsWithPushdownFilter = Seq("type", "subtype", "assetId", "startTime", "source")
+
+  override def urlsWithFilters(filters: Array[Filter], uri: Uri): Seq[Uri] = {
+    val filterMaps: Map[String, Array[String]] =
+      fieldsWithPushdownFilter.map(col => (col, filters.flatMap(getFilter(_, col)))).toMap
+    val assetIdsFilter = filterMaps.getOrElse("assetId", Array())
+    val typeFilter = filterMaps.getOrElse("type", Array())
+    val subtypeFilter = filterMaps.getOrElse("subtype", Array())
+    val sourceFilter = filterMaps.getOrElse("source", Array())
+
+    val startTimeFilters = getStartTimeFilters(filters)
+
+    val assetIds = if (assetIdsFilter.nonEmpty) {
```

Comment:
There must be a better way to do those `if (x.nonEmpty) { Seq(...) } else { Seq() }` things, I'm a bit suspicious of so many ifs and "toSeq".

Reply:
Agree, that was a bit rushed. Changed to Options + pattern matching.